### PR TITLE
Improve Python health check

### PR DESCRIPTION
**Pull request by blueyed**
- handle version from `pip install --user -e`
- report error from the `import` failure

Side notes:
- there should probably be a `neovim.VERSION` that should be used instead of the manual parsing!
- we could report the location of the found module, e.g. via `inspect.getfile()`

TODO:
 - try to import and use `neovim.VERSION`

### CODE REVIEW

**Comment by justinmk** [comment]
> there should probably be a neovim.VERSION that should be used instead of the manual parsing!

You mean for the client version? Or Nvim version? Will https://github.com/neovim/neovim/pull/5386 help?

**Comment by tweekmonster** [comment]
> Will #5386 help?

This is just for describing Neovim's API itself right?  If so, I think it'd be more helpful if the host client actually contained *its* version within the source (e.g. `neovim.__version__` for the python client).

**Comment by blueyed** [comment]
Yes, I suggest using/adding `neovim.VERSION` for this.

**Comment by bfredl** [comment]
https://github.com/neovim/python-client/pull/232 has a proposal for this. It has both `neovim.VERSION` for the version of the python module and `nvim.version` for the version of the connected nvim instance (for 0.1.6 and later).

**Comment by blueyed** [comment]
Should I do something here, apart from rebasing?

**Comment by blueyed** [comment]
@mhinz via IRC:
> Essentially you only add `.egg-info/PKG-INFO`, so it seems fine to me.

**Comment by justinmk** [comment]
Needs a rebase.

**Comment by blueyed** [comment]
Rebased, hopefully not messed up.

**Comment by blueyed** [comment]
Rebased, and added information about the `neovim` Python module's path in case of outdated/unknown versions.

**Comment by mhinz** [comment]
LGTM.

Do you mind adding another commit that removes a bit verbosity? E.g. in my case it looks like this:

```
  - INFO: Using: g:python3_host_prog = "/usr/local/bin/python3"
  - INFO: Executable: /usr/local/bin/python3
  - INFO: Python3 version: 3.6.0
  - INFO: python3-neovim version: 0.1.12
  - SUCCESS: Latest python3-neovim is installed: 0.1.12
```

Lines 2 and 4 are unneeded here.

**Comment by blueyed** [comment]
@mhinz 

    - INFO: Executable: /usr/local/bin/python3
    - INFO: Python3 version: 3.6.0

should be

    - INFO: Python3 version: 3.6.0 (from /usr/local/bin/python3)

Agreed that line 4 should be removed.

Also agreed that this should be another PR.

**Comment by blueyed** [comment]
Can somebody restart https://travis-ci.org/neovim/neovim/jobs/190045776, in case it's a temporary failure?

File: runtime/autoload/health/provider.vim [hunk 1]
```diff
@@ -375,29 +383,33 @@ function! s:check_python(version) abort
...
-    endif
-
-    if s:is_bad_response(status)
-      call health#report_warn(printf('Latest %s-neovim is NOT installed: %s',
-            \ python_bin_name, latest))
-    elseif !s:is_bad_response(latest)
-      call health#report_ok(printf('Latest %s-neovim is installed: %s',
-            \ python_bin_name, latest))
+    else
+      if s:is_bad_response(status)
```
Comment (justinmk) [inline runtime/autoload/health/provider.vim]: can avoid the extra nesting level, just make this `elseif`.

**Comment by justinmk** [comment]
@blueyed LGTM, needs one last rebase (sorry, I just pushed a Windows fix to health.vim which conflicts with this)

**Comment by blueyed** [comment]
Not sure about the conflict, I've kept HEAD:
```diff
<<<<<<< HEAD
||||||| parent of 041fe250... Improve error reporting for Python health check
      let suggestions = [
            \ 'Error found was: ' . current,
            \ 'Use the command `$ pip' . a:version . ' install neovim`',
            \ ]
=======
      let suggestions = [
            \ 'The error was: ' . status,
            \ 'To install the Python client use: `pip' . a:version . ' install neovim`',
            \ ]
>>>>>>> 041fe250... Improve error reporting for Python health check
```

**Comment by justinmk** [comment]
Great improvement, thanks @blueyed !

**Pull request merged**